pandas/core/frame.py

```diff
@@ -5411,10 +5411,17 @@ def combine(self, other, func, fill_value=None, overwrite=True):
         2  NaN  3.0 1.0
         """
         other_idxlen = len(other.index)  # save for compare
+        other_columns = other.columns
 
         this, other = self.align(other, copy=False)
         new_index = this.index
 
+        if other.empty and self.empty:
+            empty_columns = self.columns.union(other_columns)
+            for col in empty_columns:
```

Comment:
woa, you are mutating the original here, use assign instead.

Reply:
okay, will do that.